org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -175,7 +176,12 @@ public void fileChanged(String uriString, CHANGE_TYPE changeType) {
 			List<URI> uris = getURIs(settingsUrl);
 			boolean changed = false;
 			for (URI settingsURI : uris) {
-				if (settingsURI.equals(uri)) {
+				if (Platform.OS_WIN32.equals(Platform.getOS()) && URIUtil.isFileURI(settingsURI) && URIUtil.isFileURI(uri)) {
```

Comment:
this could be a helper method `compareURI(URI uri1, URI uri2)`.

Reply:
Should this util be put into JDTUtils?

Reply:
`org.eclipse.core.runtime.URIUtil#sameURI` seems to do the same thing you want. Could you check if that satisfies your needs?

Reply:
It works. I'll use it then.